lib/stat_tracker.rb

```diff
@@ -1,17 +1,23 @@
 require 'csv'
 require_relative 'team'
+require_relative 'game'
+require_relative "game_team"
 
 class StatTracker
   attr_reader :games, :teams, :game_teams
   def initialize(files)
-    @games = (CSV.open files[:games], headers: true, header_converters: :symbol).map do |row|
```

Comment:
Utilization of CSV.foreach increases memory efficiency by reading the file by individual lines. CSV.open will load the entire file into memory which may be untenable with file sizes in the hundreds of thousands of data points.